### PR TITLE
Add option to disable import-from-derivation completely, even if the drv is already realized

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by k0001** [comment]
To clarify: The rationale for this change is preventing builds from happening before intended. For example, if one has an import-from-derivation expression in some project, then when building the project in Hydra, the building of said derivation will happen at evaluation time in the Hydra host instead of happening in the build slaves at build time.

**Comment by copumpkin** [comment]
Makes sense to me!

**Comment by k0001** [comment]
We should add some documentation about this, though.

**Comment by expipiplus1** [comment]
See also #666

**Comment by shlevy** [comment]
Writing docs now

**Comment by domenkozar** [comment]
Fixes https://github.com/NixOS/nix/issues/1122

**Comment by shlevy** [comment]
Documented

File: doc/manual/command-ref/conf-file.xml [hunk 1]
```diff
@@ -644,6 +644,16 @@ password <replaceable>my-password</replaceable>
   </varlistentry>
 
 
+  <varlistentry xml:id="conf-allow-import-from-derivation"><term><literal>allow-import-from-derivation</literal></term>
+
+    <listitem><para>By default, nix allows you to <function>import</function> from a derivation,
```
Comment (edolstra) [inline doc/manual/command-ref/conf-file.xml]: nix -> Nix.

**Pull request merged**